cfgov/regulations3k/scripts/insert_section_links.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import unicode_literals
+
+import logging
+import re
+
+from regulations3k.models import Part, Section
+from regulations3k.scripts.ecfr_importer import PART_WHITELIST
+
+
+REG_BASE = '/policy-compliance/rulemaking/regulations/{}/'
+SECTION_RE = re.compile('(?:\N{SECTION SIGN}|Section|12 CFR)\W+([^\s]+)')
+PARTS_RE = re.compile(
+    r'(?P<part>\d{4})[.-](?P<section>[0-9A-Z]+)(?P<ids>\([a-zA-Z0-9)(]+)?')
+ID_RE = re.compile(r'\(([a-zA-Z0-9]{1,4})\)')
+
+logger = logging.getLogger(__name__)
+
+
+def get_url(section_reference):
+    if not PARTS_RE.match(section_reference):
+        return
+    parts = PARTS_RE.match(section_reference).groupdict()
+    part = parts.get('part')
+    if part not in PART_WHITELIST:
+        return
+    part_url = REG_BASE.format(part)
+    section_url = "{}{}/".format(part_url, parts.get('section'))
+    if not parts.get('ids'):
+        return section_url
+    paragraph_id = "-".join(ID_RE.findall(parts.get('ids')))
+    if paragraph_id:
+        return "{}#{}".format(section_url, paragraph_id)
+    else:
+        return section_url
+
+
+def insert_section_links(regdown):
+    """Turn internal section references into links."""
+    section_refs = SECTION_RE.findall(regdown)
+    if not section_refs:
+        return
+    index_head = 0
+    for i, ref in enumerate(section_refs):
+        url = get_url(ref)
+        if url:
+            link = '<a href="{}" data-linktag="{}">{}</a> '.format(url, i, ref)
```

Comment:
Should it be an `a` tag or a Markdown link?

Reply:
For better or worse, I wanted script-added links to be unique and identifiable, so that the script wouldn't operate twice on the same reference if run again, and to provide a hook in case we wanted to strip out script-added links without touching editor-added markdown links.

The tags aren't as editor-friendly, but tag pollution could be undone.